### PR TITLE
21 push notification

### LINKED ISSUE #21: [21pts] Reminders/Push Notifications
**Fill in the blanks**:
As a student, I would like to get reminded when my favorite foods are being served, so that I can prioritize going to DCT over other dining options on certain days.

**Definition of Done - Check as you go!**
- [ ] Tasks have been implemented
- [ ] Documented
- [ ] Passing Tests for each task
- [ ] Pull Request made
- [ ] Code Reviewed

### PR DESCRIPTION
**Pull request by DDVD233**
It can now add a device to a user's device list and notify the user whenever a reminder is required.

Several todos:
- We need a scheduler to send reminders periodically, as sending reminders are not invoked by accessing an API endpoint. 
- Frontend needs to request notification permission. We need to have a somewhat friendly interface to ask user for permission. 

The todos can be another ticket, separated from this PR to make the PR not too large.

### CODE REVIEW

File: app_backend/models/user.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        The email of the user
+    password: str
+        The password of the user
+"""
+class User(BaseModel):
+    id: PyObjectId = Field(default_factory=PyObjectId, alias="_id")
+    name: str
+    email: str
+    password: str
+    devices: Optional[list]
```
Comment (RafaelPiloto10) [inline app_backend/models/user.py]: Not implemented yet, but nice to have so we can get this merged ASAP and help other PR's is Dooley $ and meal swipes

**Review by RafaelPiloto10 — APPROVED** [review]
This PR is top priority, hopefully we can get it merged ASAP. Just a minor thing in preparation of other PR's

File: app_backend/models/user.py (hunk 1 above)
Comment (RafaelPiloto10) [reply app_backend/models/user.py]: Also friends (list of uuids), and liked foods (list of strings?), station (string), is_sharing_location (boolean)

Comment (DDVD233) [reply app_backend/models/user.py]: OK will add these

**Comment by DDVD233** [comment]
I am closing this PR and opening a new one as the "file changed" does not show the right number.

**Pull request closed without merging**